server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -588,6 +588,17 @@ public void addResponseHeader(final String key, final String value) {
         addResponseHeader(key, value, v -> v);
     }
 
+    /**
+     * Add the {@code value} for the specified {@code key}. Any duplicate {@code value} is ignored.
+     * True values are converted to "true", and false to "false".
+     *
+     * @param key         the header name
+     * @param value       the header boolean value
+     */
+    public void addResponseHeader(final String key, final boolean value) {
+        addResponseHeader(key, value ? "true" : "false", v -> v);
```

Comment:
I wonder if we have another standard to follow for boolean values

Reply:
I thought that doing it this way would avoid having different formats for bools in the future. Having this method as the "formatter for booleans".
But this being an uncommon thing, it works for me. Changing it!

Reply:
FWIW [RFC 8941 §3.3.6](https://datatracker.ietf.org/doc/html/rfc8941#section-3.3.6) specifies the strings `?0` and `?1` to represent `false` and `true` respectively. It's our header, we can do what we want, just thought I'd mention it.

Reply:
I hate to deviate from the RFC, but a quick scan doesn't say why they can't use true or false. OTOH I'm used to the "present means true" semantics from request parameters.

To be honest, maybe we can dodge the whole thing and just return the id if we're still running and it's presence means we're running and it's absence means where not.

Reply:
It means that you can unambiguously distinguish the types `boolean` and `token`, and that this distinction only needs you to look at the first byte. But yeah maybe we don't even need this.

Reply:
About just having the ID, it would work. However, it could be confusing if querying older versions of ES (by mistake) (No header == not running VS No header at all). I'm not sure we care much about this, but I find it more error prone (?)

If everybody's fine with it, changing to ?0/1